spinnaker/spinnaker.go

```diff
@@ -495,32 +518,33 @@ func (s Spinnaker) CloudProvider(account string) (provider string, err error) {
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", errors.Wrap(err, fmt.Sprintf("body read failed at %s", url))
-	}
-
-	var fields struct {
-		CloudProvider string `json:"cloudProvider"`
-		Error         string `json:"error"`
+		return nil, errors.Wrap(err, fmt.Sprintf("body read failed at %s", url))
 	}
 
-	err = json.Unmarshal(body, &fields)
+	accounts := make([]account, 0)
+	err = json.Unmarshal(body, &accounts)
 	if err != nil {
-		return "", errors.Wrap(err, "json unmarshal failed")
+		return nil, errors.Wrap(err, "json unmarshal failed")
 	}
+	statusKO := resp.StatusCode != http.StatusOK
 
-	if resp.StatusCode != http.StatusOK {
-		if fields.Error == "" {
-			return "", fmt.Errorf("unexpected status code: %d. body: %s", resp.StatusCode, body)
+	// Finally find account
+	for _, a := range accounts {
+		if a.Name != name {
+			continue
 		}
+		if statusKO {
+			if a.Error == "" {
+				return nil, fmt.Errorf("unexpected status code: %d. body: %s", resp.StatusCode, body)
```

Comment:
suggest to be consistent in errors creation.
Use errors.Errorf instead
https://godoc.org/github.com/pkg/errors#Errorf